django/thunderstore/repository/validators.py

```diff
@@ -60,3 +60,8 @@ def __eq__(self, other):
     regex=AUTHOR_NAME_REGEX,
     message="Author names can only contain a-Z A-Z 0-9 . _ - characers",
 )
+
+
+def license_validator(value: str) -> None:
+    if value not in SPDX_LICENSE_IDS:
```

Comment:
We should probably allow an empty string for licenses that aren't listed in SPDX

Reply:
What do you mean?

Reply:
Allow not defining a license, but keep it a required field. This would mean the user has to explicitly define they don't have a license (rather than just accidentally omitting the field), or if they ship a license that has an SPDX identifier.

Reply:
Point is I don't think we can require every package to be licensed under some license found in SPDX

Reply:
Would `"None"` (or similar) work? That way it is clear from reading the manifest that they have no license instead of it being some default in a manifest template

Reply:
`"None"` added in dd8a61b6bcc1dca9bdb9e8a7d2a7c539e587e8d2

Reply:
Not having an SPDX identifiable license doesn't mean it doesn't have a license though, it just means there's no SPDX identifier for it. Thus I think it's clearer to have it as an empty string, as it leaves it ambiguous: there may or may not be a license, but we don't know what it is.

Reply:
So does `""` just mean "we don't know"? Should I change the default value in the migration to `""` too? Or is there a use for differentiating between `No License` and unknown license

Reply:
Yeah that's correct, empty means we don't know what it is, AKA undefined. It just makes more sense requiring users to explicitly define a license field to be empty, since it avoids accidents where they simply forgot to add a license.

Reply:
By default if you don't grant a license for something you've built, it's all rights reserved for you, essentially making that our "default". Your package however might for example have a LICENSE file with a custom license, we just can't pick it up automatically (at least as of now)